addons/point_of_sale/models/pos_order.py

```diff
@@ -1040,7 +1043,7 @@ def sync_from_ui(self, orders):
                 raise ValidationError(_('You can only refund products from the same order.'))
 
             existing_order = self._get_open_order(order)
-            if existing_order and existing_order.state == 'draft':
+            if existing_order:
```

Comment:
```suggestion
            order_ids.append(self._process_order(order, existing_order or False))
            _logger.info("PoS synchronisation #%d order %s %s pos.order #%d", sync_token, order_log_name, 'updated' if existing_order else 'created', order_ids[-1])
```

I think you can remove line 1046 to 1056 and replace them by this